src/ado/reprun.ado

```diff
@@ -260,8 +279,24 @@ end
           local line_command = "OTHER"
           local dofile ""
           local doflag 0
-          foreach w in `macval(line)' {
-            get_command, word("`w'")
+
+          local theline = regexr(`"`macval(line)'"',"[\[//]\\\^\%\|\?\*\+\(\)]","")
+          forv i = 1/10 {
+            local theline = regexr(`"`theline'"',"[\[//]]","")
+            local theline = regexr(`"`theline'"',"[\\]","")
+            local theline = regexr(`"`theline'"',"[\^]","")
+            local theline = regexr(`"`theline'"',"[\%]","")
+            local theline = regexr(`"`theline'"',"[\|]","")
+            local theline = regexr(`"`theline'"',"[\?]","")
+            local theline = regexr(`"`theline'"',"[\*]","")
+            local theline = regexr(`"`theline'"',"[\+]","")
+            local theline = regexr(`"`theline'"',"[\(]","")
+            local theline = regexr(`"`theline'"',"[\)]","")
+            // local theline = regexr(`"`theline'"',`"[\"]"',"")
+          }
+
+          foreach w in `macval(theline)' {
+            cap get_command, word(`"`w'"')
```

Comment:
This section is attempting to sanitize strings to pass into `get_command`. It works decently should rather be a `while` loop most likely, and should probably put in spaces instead of blanks.

However, I can't figure out an appropriate handling for `"`. We cannot have unmatched `"` or the `foreach` line will break. At the same time, we cannot remove `"` because they are also needed to delineate words.

One solution @luisesanmartin proposed is to have a better handling for catching comments so they can be skipped here entirely. Another solution is to simply note failures here and fix them manually -- they are somewhat rare but @ankritisingh has encountered a couple (especially in comments, which is particularly painful).

Another solution is to take the handling of dofiles out of this loop, since we are only trying to catch `do`/`run` here. We could handle that separately in the full line if desired, rather than trying to catch the next word at this point.

Reply:
@kbjarkefur

Reply:
This was the latest one but I don't see an unmatched `"` 

Error:

![Screenshot 2024-06-06 at 4 26 48 PM](https://github.com/worldbank/repkit/assets/54277703/f0ae0e58-0ad0-41f9-95ef-04025515c45a)

Code:

![Screenshot 2024-06-06 at 4 26 43 PM](https://github.com/worldbank/repkit/assets/54277703/4a4ead48-be3d-42c8-ad3b-538e090b93a2)

Reply:
This is the unhandled `*`. Hence needing a `while` loop to catch them all. @luisesanmartin also interesting that catching comments won't help here!

Reply:
Also, it's possible that it's taking `cmd"pdslasso` alone as a word, which will still produce a lone `"` in the next step.